addons/mail/wizard/mail_template_preview.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+LIST_FIELDS_MAIL_TEMPLATE = ['subject', 'body_html', 'email_from', 'email_to',
+                             'email_cc', 'reply_to', 'scheduled_date', 'attachment_ids']
+
+
+class MailTemplatePreview(models.TransientModel):
+    _name = 'mail.template.preview'
+    _description = 'Email Template Preview'
+
+    @api.model
+    def _selection_target_model(self):
+        return [(model.model, model.name) for model in self.env['ir.model'].search([])]
+
+    @api.model
+    def _selection_languages(self):
+        return self.env['res.lang'].get_installed()
+
+    @api.model
+    def default_get(self, fields):
+        result = super(MailTemplatePreview, self).default_get(fields)
+        default_mail_template_id = self.env.context.get('default_mail_template_id')
+        if not default_mail_template_id or 'resource_ref' not in fields:
+            return result
+        mail_template = self.env['mail.template'].browse(default_mail_template_id)
+        res = self.env[mail_template.model_id.model].search([], limit=1)
+        if res:
+            result['resource_ref'] = '%s,%s' % (mail_template.model_id.model, res.id)
+        return result
+
+    mail_template_id = fields.Many2one('mail.template', string='Related Mail Template', required=True)
+    model_id = fields.Many2one('ir.model', string='Targeted model', related="mail_template_id.model_id")
+    resource_ref = fields.Reference(string='Record', selection='_selection_target_model')
+    lang = fields.Selection(_selection_languages, string='Template Preview Language')
+    error_msg = fields.Char('Error Message', readonly=True)
+    # Fields same than the mail.template model, computed with resource_ref and lang
+    subject = fields.Char('Subject', compute='_compute_mail_template_fields')
+    email_from = fields.Char('From', compute='_compute_mail_template_fields', help="Sender address")
+    partner_ids = fields.Many2many('res.partner', string='Recipients', compute='_compute_mail_template_fields')
+    email_to = fields.Char('To (Emails)', compute='_compute_mail_template_fields',
```

Comment:
Small detail:

You can remove the " (Emails)" part and just keep "To", it's a little confusing when previewing.

(It's done like that in mail.template model because there is also a "partner_to" field, but it's not the case in the preview model.)

Reply:
Alright, done!